packages/wallet/src/signer.ts

```diff
@@ -1,16 +1,21 @@
 import { NetworkConfig } from '@0xsequence/network'
-import { Transactionish } from '@0xsequence/transactions'
-import { JsonRpcProvider, TransactionResponse } from '@ethersproject/providers'
+import { Relayer } from '@0xsequence/relayer'
+import { SequenceTransaction, SignedTransactions, Transactionish } from '@0xsequence/transactions'
+import { Provider, TransactionResponse } from '@ethersproject/providers'
 import { BigNumberish, Signer as AbstractSigner } from 'ethers'
 import { BytesLike, Deferrable } from 'ethers/lib/utils'
 import { WalletConfig } from '.'
 
 export abstract class Signer extends AbstractSigner {
-  readonly provider?: JsonRpcProvider
+  abstract getProvider(chainId?: number): Promise<Provider | undefined>
```

Comment:
can we return Promise<JsonRpcProvider | undefined> instead? I think will be easier to consume and why not use JsonRpcProvider for all?

Reply:
btw, JsonRpcProvider is also of type Provider, so if something requires Provider, it will still be accepted.